NicolaeChristof/Rise
# Enhancement revised branch ui

This integrates Malcolm's branch provider and UIBranchElement into the UIController.cs file. It also has a few miscellaneous changes (like fixing the height UI not updating correctly and changing some branch colors so they fit the UI branches).

Note:
Because I'm currently using a keyboard, I let you have split-screen while using a keyboard. This is just to test it, and the code to enforce single-screen on keyboard is still in the settings controller. There's a variable I added in SettingsController.cs called enforceModes (based on the name Malcolm gave it) that you can turn on/off to remove/add the splitscreen functionality to keyboard.

How to Test:
Try flipping through the different menus (except the ones that say unimplemented) and making sure the height UI/branches only appear where they're supposed to. Also ensure that restarting doesn't mess with the branch UI. I wasn't able to extensively test out multiplayer, so that also is a good place to look at.
